lib/active_merchant/billing/gateways/global_collect.rb

```diff
@@ -289,8 +291,10 @@ def error_code_from(succeeded, response)
         unless succeeded
           if errors = response["errors"]
             errors.first.try(:[], "code")
+          elsif status = response["statusOutput"]["statusCode"]
```

Comment:
> I'm not sure I see how it would hit an exception in that case.

So, `response["statusOutput"]` would be `nil` if that key doesn't exist at the top level of the response hash. Which means it'd be the same as `nil["statusCode"]` which would raise the exception.

> The first if is in the same position.

Fortunately this is slightly different since if you reference a nonexistent key in a hash, `nil` is returned-

```ruby
➜ irb
irb(main):001:0> hash = {a: 1}
=> {:a=>1}
irb(main):002:0> hash[:b]
=> nil
```

> kind of wondering why I don't see it in more places.

I've been using it anytime I'm going more than one level down in a hash. Purely for the sake of avoiding a no method error on `nil` since getting nil value is better than an exception being raised in the middle of parsing out a transaction response.